### ISSUE TITLE
Cannot create catalog with vcd admin user

### ISSUE THREAD

**Issue by TheEric7** [vmware/ansible-module-vcloud-director#172]
Can't create a catalog with vCloud Director's Admin user which has access to all organizations.
My playbook is:
```
---
- name: create catalog
  hosts: localhost
  environment:
    env_user: my_user
    env_password: my_password
    env_host: my_host
    env_org: system
    env_api_version: 30.0
    env_verify_ssl_certs: false
  tasks:
  - name: create catalog
    vcd_catalog:
      catalog_name: "test"
      description: "test"
      state: "present"
```

And the expected error is **"Status code: 400/BAD_REQUEST, No storage available in the organization. Please allocate resources to your organization.** This is understandable because the catalog should be created on an organization other than system. But when I add the org name it says can't login to the organization with that user, even though that user is Admin and has access to every organization.

```
---
- name: create catalog
  hosts: localhost
  environment:
    env_user: my_user
    env_password: my_password
    env_host: my_host
    env_org: system
    env_api_version: 30.0
    env_verify_ssl_certs: false
  tasks:
  - name: create catalog
    vcd_catalog:
      catalog_name: "test"
      description: "test"
      state: "present"
      org: "my-org"    <<<<<<<
```
And the error is:
**"Login failed for user my_user to org my-org"**

Note that I'm able to work with some other modules without issues this way. i.e.
```
---
- name: create org vdc direct network
  environment:
    env_user: my_user
    env_password: my_password
    env_host: my_host
    env_org: system
    env_api_version: 30.0
    env_verify_ssl_certs: false
  tasks:
  - name: create network
    vcd_vdc_network:
      org_name: test-org
      vdc_name: test-vdc
      network_name: test-network
      parent_network_name: test-network
      shared: false
      direct: true
      state: present
```

**Comment by mukultaneja** [comment]
@TheEric7, that is correct. Your first query regarding not having access, that is the expected behaviour and about the second query, I think we should not be having `org_name` as part of `vcd_vdc_network`. I can not think of any use-case for this argument, do you suggest any use-case here?

**Comment by TheEric7** [comment]
I don't have any use case for the second one as I just wanted to tell you I have even tested that way as well. If you are saying that what I see in the first query is kind of expected, then I have to change my approach.

My understanding was that if you have an account that is VCD Admin, then you have access to all organizations and can change the settings as if you are the local org admin. But what you are saying is that the first output is expected, which means that you should have a local org admin, is that right? If that's the case, for what actions do I need to have local org admin users? Because as I mentioned, I am able to do the majority of my tasks with the VCD admin user and I didn't know there are some which need local org admin.

**Comment by TheEric7** [comment]
I did a bit of reading and I believe my understanding about **"if you have an account that is VCD Admin, then you have access to all organizations and can change the settings"** was not right.
Simply speaking, whatever resource you have on the VDC needs to have an owner, and that local admin user on the VDC will be the owner, thus using a local admin user in the module completely makes sense.
Closing the case.
